src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0020Tests.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+using Verifier = Azure.ClientSdk.Analyzers.Tests.AzureAnalyzerVerifier<Azure.ClientSdk.Analyzers.BannedTypesAnalyzer>;
+
+namespace Azure.ClientSdk.Analyzers.Tests
+{
+    public class AZC0020Tests
+    {
+        private List<(string fileName, string source)> _sharedSourceFiles;
+
+        public AZC0020Tests()
+        {
+            _sharedSourceFiles = new List<(string fileName, string source)>() {
+
+            ("MutableJsonDocument.cs", @"
+                namespace Azure.Core.Json
+                {
+                    internal sealed partial class MutableJsonDocument
+                    {
+                    }
+                }
+                "),
+
+            ("MutableJsonElement.cs", @"
+                namespace Azure.Core.Json
+                {
+                    internal sealed partial class MutableJsonElement
+                    {
+                    }
+                }
+                ")
+            };
+        }
+
+        [Fact]
+        public async Task AZC0020ProducedForMutableJsonDocumentUsage()
+        {
+            string code = @"
+using Azure.Core.Json;
+
+namespace LibraryNamespace
+{
+    public class Model
+    {
+        MutableJsonDocument {|AZC0020:_document|};
+    }
+}";
+            await Verifier.VerifyAnalyzerAsync(code, _sharedSourceFiles);
+        }
+
+        [Fact]
+        public async Task AZC0020ProducedForMutableJsonElementUsage()
+        {
+            string code = @"
+using Azure.Core.Json;
+
+namespace LibraryNamespace
+{
+    public class Model
+    {
+        MutableJsonElement {|AZC0020:_element|};
+    }
+}";
+            await Verifier.VerifyAnalyzerAsync(code, _sharedSourceFiles);
```

Comment:
If you are saying we cannot make `_element` public, then I think the private one should be a positive case.

Reply:
> How could we fail this?

Yes, good question!  The idea is that we will suppress AZC0020 for generated Patch models only, until we move the MutableJsonDocument types into the Azure.Core public API.  It is to prevent people from using them for purposes we don't know about, so they don't take dependencies on functionality that we plan to change.  There are more details about this in this comment: https://github.com/Azure/azure-sdk-tools/pull/6642#issue-1832206874

Reply:
> If you are saying we cannot make _element public, then I think the private one should be a positive case.

This analyzer should prevent its use for any case outside of the Azure.Core library.